docs/content/basics/ui_testing.md

```diff
@@ -1,4 +1,148 @@
 # UI testing
 
-To be done
+UI tests are a part of instrumentation tests. <br/>
+That's why everything from [Instrumented tests topic](https://android-ui-testing.github.io/Cookbook/basics/instrumented_testing_basics/) is applicable to UI testing.
 
+Ui testing goal is to check some scenarios of using your application.<br/>
+Usually, we do ui testing for some common users scenario:
+
+* Login to your application
+* Process of ordering 
+* Creating new chat 
+* And so on
+
+Simple Ui test can't catch the problem with wrong padding or wrong color of your view (only in cases where you specially check that) - because there is another type of tests used to catch such problems called `Screenshot testing` - you can read [Screenshot testing article](https://android-ui-testing.github.io/Cookbook/basics/screenshot_testing/) to get familiar with that.<br/>
+
+So, if once we decided that we need to check users scenarios - we should definitely start writing Ui tests and let's start with tools, that will help us to write this type of tests.
+
+## Main tool - Espresso and UiAutomator
+
+Nowadays there is no doubt - for native UI testing on Android we should use Espresso for testing your application.
+<br/>This is a main tool that allow us to make every test possible to access your codebase. That means tests with Espresso allow us to write white-box tests.
+<br/>But you still able to write black-box tests or even gray-box with Espresso.
+
+Why Espresso is so cool? It is synchronized with the main thread of your app and performs action only when main thread is in idle state.
+<br/>It makes Espresso extremely reliable and stable (than any other existing tool) for Ui testing on Android.
+
+Keep in mind that everything that isn't related to your app (for example, permission dialogs, volume +/- buttons, home button and etc) - can't be accessed directly within Espresso.
+<br/>But don't worry - you can use `UiAutomator` for that purpose, and it can be called directly from test, written in Espresso.
+<br/>`UiAutomator` is a slower tool and can support tests only in black-box style, but it can test everything outside your application on your phone.
+<br/>Many cross-platforms tools for testing using `UiAutomator`.
+
+These two tools are the main tools that everyone using for Ui testing on Android. <br/>
+Remember, it's possible to create tests that will use both frameworks. For example, you can find your views and interact with them by `Espresso` and in the same test call `UiAutomator` to perform something specific, like pressing volume button. <br/>
+But for usual testing you should prefer `Espresso` rather then `UiAutomator`. 
+
+### Espresso
+
```

Comment:
Maybe also worth mentioning that Espresso as well as the Jetpack Compose Library support Robolectric, while UiAutomator doesn't.

That means, one can write Shared Tests written in Espresso or with Compose Library